modules/platforms/cpp/thin-client/include/ignite/thin/transactions/transactions.h

```diff
@@ -0,0 +1,191 @@
+             * create another reference to the same underlying object. Underlying object released automatically once all
+             * the instances are destructed.
+             */
+            class ClientTransactions {
+            public:
+                /**
+                 * Constructor.
+                 *
+                 * @param impl Implementation.
+                 */
+                ClientTransactions(SharedPointer<void> impl) :
+                    proxy(impl),
+                    label(SharedPointer<const char>())
+                {
+                }
+
+                /**
+                 * Destructor.
+                 */
+                ~ClientTransactions()
+                {
+                }
+
+                /**
+                 * Start new transaction with completely clarify parameters.
+                 *
+                 * @param concurrency Transaction concurrency.
+                 * @param isolation Transaction isolation.
+                 * @param timeout Transaction timeout.
+                 * @param txSize Number of entries participating in transaction (may be approximate).
+                 *
+                 * @return ClientTransaction implementation.
+                 */
+                ClientTransaction TxStart(
+                        TransactionConcurrency::Type concurrency = TransactionConcurrency::PESSIMISTIC,
+                        TransactionIsolation::Type isolation = TransactionIsolation::READ_COMMITTED,
+                        int64_t timeout = 0,
+                        int32_t txSize = 0)
+                {
+                    return ClientTransaction(proxy.txStart(concurrency, isolation, timeout, txSize, label));
+                }
+
+                /**
+                 * Returns instance of {@code ClientTransactions} to mark each new transaction with a specified label.
+                 *
+                 * @param label Transaction label.
+                 * @return ClientTransactions implementation.
+                 */
+                ClientTransactions withLabel(const std::string& lbl)
+                {
+                    ClientTransactions copy = ClientTransactions(proxy, lbl.c_str());
+
+                    return copy;
+                }
+            private:
+                /** Implementation. */
+                TransactionsProxy proxy;
+
+                /** Transaction specific label. */
+                SharedPointer<const char> label;
+
+                /**
+                 * Default constructor.
+                 */
+                ClientTransactions();
+
+                /**
+                 * Constructor.
+                 *
+                 * @param impl Implementation.
+                 */
+                ClientTransactions(TransactionsProxy& impl, const std::string& lbl) :
+                    proxy(impl)
+                {
+                    char* l = new char[lbl.size() + 1];
```

Comment:
`SharedPointer` uses `delete` as a deleter by default, not `delete[]`. Use something different for a container here, for example `SharedPointer<common::FixedSizeArray<char>>`.